File: Python/queues.py

class Node:
    def __init__(self,data=None,next=None):
            self.data=data
            self.next=next

class queue:
    def __init__(self,head=None,tail=None):
        self.head=head
        self.tail=tail
    def isEmpty(self):
        return self.head==None
    def peak(self):
        return self.head.data

    def add(self,data):
        node=Node(data=data)
        if self.tail!=None:
            self.tail.next=node
        self.tail=node
        if self.head==None:
            self.head=node
    def remove(self):
        data=self.head.data
        self.head=self.head.next
        if(self.head==None):
            self.tail=None
        return data

def main():
    new_queue=queue()
    new_queue.add(5)
    new_queue.add(2)
    print(new_queue.remove())
    print("The new head is {}".format( new_queue.peak()))

if __name__ == "__main__":
    main()

File: C++/queues.cpp
#include <iostream>
#include <vector>
#include "Node.h"
using namespace std;




class Queue{ 
    public:
   
    Node head=Node();
    Node tail=Node();
    Queue(void);
        int peak(){
            return head.data;
        }
        void add (int data){
            Node node=Node(data);
            if (!tail.isEmpty()){
                tail.next=node;
            }
            tail=node;

            if (head.isEmpty()){
                head=node;
            }

        }
        int remove(){
            int data=head.data;
            head=head.next;
            if(head.isEmpty()){
                tail=Node();
            }
            return data;
        }

    };



int main(){
   
    Queue queue=Queue();
    queue.add(5);
    queue.add(2);
    queue.remove();
    cout << queue.peak();

}